### PR TITLE
Add workaround flag for lazy import-as-member

### PR DESCRIPTION
**Pull request by beccadax**
In rdar://123649082, a project failed to build because of the lazy import-as-member loading changes in #71320. That project was configured in a way that broke modularization and the correct solution is to fix it, but out of an abundance of caution, add a `-disable-named-lazy-import-as-member-loading` frontend flag in case a project needs to temporarily restore the old behavior.

As a bonus, this lets us write a test to verify that lazy import-as-member loading has positive performance impact.

This PR also includes a frontend flag that dumps ClangImporter's SwiftLookupTables on frontend process termination; this was previously possibly only with a debugger or using swift-ide-test.

Fixes rdar://124228983.

### CODE REVIEW

**Comment by beccadax** [comment]
@swift-ci test

File: test/IDE/dump_swift_lookup_tables.swift [hunk 1]
```diff
@@ -1,5 +1,9 @@
-// RUN: %target-swift-ide-test -dump-importer-lookup-table -source-filename %s -import-objc-header %S/Inputs/swift_name.h -I %S/Inputs/custom-modules > %t.log 2>&1
-// RUN: %FileCheck %s < %t.log
+// RUN: %target-swift-ide-test -dump-importer-lookup-table -source-filename %s -import-objc-header %S/Inputs/swift_name.h -I %S/Inputs/custom-modules > %t.ide-test.log 2>&1
+// RUN: %FileCheck %s < %t.ide-test.log
+
+// RUN: %target-typecheck-verify-swift -dump-clang-lookup-tables -import-objc-header %S/Inputs/swift_name.h -I %S/Inputs/custom-modules > %t.frontend.log 2>&1
+// RUN: %FileCheck %s < %t.frontend.log
+// RUN: false
```
Comment (ahoppen) [inline test/IDE/dump_swift_lookup_tables.swift:6]: Is this a debug leftover?

Comment (beccadax) [reply test/IDE/dump_swift_lookup_tables.swift:6]: Yes. Yes, it is.

**Comment by glessard** [comment]
@swift-ci please test

**Comment by beccadax** [comment]
@swift-ci test

**Pull request merged**